File: Week_1/Cash_Credit/Cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

int get_cents(void)
{
    int i;
    do
    {
        i = get_int("Change owed: ");
    }
    while (i < 0);
    return i;
}

//calculating quarters
int calculate_quarters(int cents)
{
    int result;
    if (cents >= 25)
    {
        result = cents / 25;
    }
    else
    {
        result = 0;
    }
    return result;
}

//calculating dimes
int calculate_dimes(int cents)
{
    int result;
    if (cents >= 10)
    {
        result = cents / 10;
    }
    else
    {
        result = 0;
    }
    return result;
}

//calculating nickels
int calculate_nickels(int cents)
{
    int result;
    if (cents >= 5)
    {
        result = cents / 5;
    }
    else
    {
        result = 0;
    }
    return result;
}

//calculating pennies
int calculate_pennies(int cents)
{
    int result;
    if (cents >= 1)
    {
        result = cents / 1;
    }
    else
    {
        result = 0;
    }
    return result;
}

File: Week_6/sentimental-cash/cash.py
# TODO
from cs50 import get_float


def get_cents():
    while True:
        i = get_float("Change owed: ")
        if (i > 0):
            break
    coin = int(i * 100)
    # print(coin)
    return coin


def calculate_quarters(cents):
    if (cents >= 25):
        result = cents / 25
    else:
        result = 0
    return int(result)


def calculate_dimes(cents):
    if (cents >= 10):
        result = cents / 10
    else:
        result = 0
    return int(result)


def calculate_nickels(cents):
    if (cents >= 5):
        result = cents / 5
    else:
        result = 0
    return int(result)


def calculate_pennies(cents):
    if (cents >= 1):
        result = cents / 1
    else:
        result = 0
    return int(result)


cents = get_cents()
# print(cents)

quarters = calculate_quarters(cents)
cents = cents - quarters * 25
# print(quarters)
# print(cents)

dimes = calculate_dimes(cents)
cents = cents - dimes * 10

nickels = calculate_nickels(cents)
cents = cents - nickels * 5

pennies = calculate_pennies(cents)
cents = cents - pennies * 1

coins = quarters + dimes + nickels + pennies
print(coins)